src/string_vector_serializer.cpp

```diff
@@ -0,0 +1,52 @@
+// See the file "COPYING" in the main distribution directory for copyright.
```

Comment:
General comment on this implementation: It's copying the strings quite a bit during serialization & unserialization. Given that these vectors could be large, I think it's worth optimizing it a bit more and probably also just return a raw memory block and return it by pointer. How about this for serialization: we could represent each string as byte sequence `<length><data>` and concatenate them all inside a chunk of memory. The size of that chunk we can precompute and allocate at once, then we just copy the data in there one by one. Then we return a `std::unique_ptr<void *>` to this thing. Or, maybe more modern: `std::unique_ptr<std::vector<std::byte>>`.

Reply:
I like that idea a lot. I'll go ahead and work on that today. Using the vector seems pretty reasonable and probably reduces the risk that I end up writing code that leaks memory or otherwise messes up. Really appreciate the input as always.